### PR TITLE
[12.0][ADD] helpdesk_mgmt_timesheet_restricted

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Adds group that cannot see the times in the timesheet and tickets.

### CODE REVIEW

File: helpdesk_mgmt_timesheet_restricted/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Helpdesk Group Timesheet Restricted",
+    "summary": """
+       Adds group that cannot see the times in the timesheet and tickets.
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/__manifest__.py]: Better this IMO:
```suggestion
       Adds group that cannot see time data in the timesheet and tickets.
```

File: helpdesk_mgmt_timesheet_restricted/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    """,
+    "version": "12.0.1.0.0",
+    "license": "LGPL-3",
+    "category": "After-Sales",
+    "author": "Solvos",
+    "website": "https://github.com/solvosci/slv-helpdesk",
+    "depends": [
+        "helpdesk_type",
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/__manifest__.py]: This dependency is not required

File: helpdesk_mgmt_timesheet_restricted/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* David Alonso <david.alonso@solvos.es>
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/readme/CONTRIBUTORS.rst]: Maybe you should add yourself as contributor 😆

File: helpdesk_mgmt_timesheet_restricted/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Adds group that cannot see the times in the timesheet and tickets.
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/readme/DESCRIPTION.rst]: The same as above

File: helpdesk_mgmt_timesheet_restricted/security/helpdesk_mgmt_timesheet_restricted.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
+        <record id="group_helpdesk_timesheet_restricted" model="res.groups">
+            <field name="name">Timesheet Restricted</field>
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/security/helpdesk_mgmt_timesheet_restricted.xml]: More accurate:
```suggestion
            <field name="name">Don't show time data in timesheets and tickets</field>
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review, seems to correctly fit to my requirements. See comments anyway.

Also, try to add a `readme/USAGE.rst` file with a few instructions for this addon, like https://github.com/OCA/helpdesk/blob/12.0/helpdesk_motive/readme/USAGE.rst , referring to how to activate it for a certain user

File: helpdesk_mgmt_timesheet_restricted/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,4 @@
+* Go to Settings -> Users & Companies -> Users
+* Select the user
+* In the Helpdesk & Timesheet section 
+* Mark "Don't show time data in timesheets and tickets"
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/readme/USAGE.rst]: This should be one line:
```suggestion
* In the Helpdesk & Timesheet section mark "Don't show time data in timesheets and tickets"
```

File: helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+<odoo>
+    <record id="view_helpdesk_ticket_timesheet_restricted_view_tree_inherit" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.tree (in helpdesk_mgmt_timesheet_restricted)</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt_timesheet.ticket_view_tree"/>
+        <field name="groups_id" eval="[(4,ref('group_helpdesk_timesheet_restricted'))]"/>
+        <field name="arch" type="xml">
+            <!--xpath expr="//field[@widget='float_time']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath-->
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml]: Remove unused code

File: helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    <record id="view_helpdesk_ticket_timesheet_restricted_view_form_inherit" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.form (in helpdesk_mgmt_timesheet_restricted)</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt_timesheet.timesheet_helpdesk_ticket_view_form"/>
+        <field name="groups_id" eval="[(4,ref('group_helpdesk_timesheet_restricted'))]"/>
+        <field name="arch" type="xml">
+            <!--xpath expr="//field[@widget='float_time']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath-->
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml]: Remove unused code

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments. Also, `progress` field for a ticket should be hidden.

File: helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+    <record id="view_helpdesk_ticket_timesheet_restricted_view_form_inherit" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.form (in helpdesk_mgmt_timesheet_restricted)</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt_timesheet.timesheet_helpdesk_ticket_view_form"/>
+        <field name="groups_id" eval="[(4,ref('group_helpdesk_timesheet_restricted'))]"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page/group" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath>
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml]: Unfortunately, `page` element on parent view has no `name` attribute. Selecting without attributes is dangerous, so I prefer this:
```suggestion
            <xpath expr="//field[@name='planned_hours']" position="attributes">
                <attribute name="invisible">1</attribute>
            </xpath>
            <xpath expr="//field[@name='progress']" position="attributes">
                <attribute name="invisible">1</attribute>
            </xpath>
```
It's more code and result is slightly different, but more secure.

**Review by dalonsod — CHANGES_REQUESTED** [review]

File: helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml (hunk 9 above)
Comment (ChristianSantamaria) [reply helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml]: @dalonsod You are right, my first solution was that, but there is an empty space that I did not like. The problem is the group. But if you like that me too.
imagen

Comment (dalonsod) [reply helpdesk_mgmt_timesheet_restricted/views/helpdesk_mgmt_timesheet_restricted.xml]: I'm aware, please do it

**Review by dalonsod — APPROVED** [review]
👍 
Please squash commits

**Comment by dalonsod** [comment]
Merging...

**Pull request merged**